src/tools/cephfs/first-damage.py

```diff
@@ -84,8 +84,8 @@ def traverse(MEMO, ioctx):
                 ioctx.operate_read_op(rctx, o.key)
                 nkey = None
                 for (dnk, val) in it:
-                    log.debug(f'\t{dnk}: val size {len(val)}')
-                    (first,) = struct.unpack('<I', val[:4])
+                    (first,) = struct.unpack('<Q', val[:8])
+                    log.debug(f'\t{dnk}: val size {len(val)} first {first}')
```

Comment:
let's keep the original log.debug as it is since if we run into unpack issues, nothing gets logged (so, log the first field after unpacking).

Reply:
Done

Reply:
It's also a 64 bit integer
```
kotresh:src$ git grep CEPH_NOSNAP | grep define | grep rados
include/rados.h:#define CEPH_NOSNAP  ((__u64)(-2))  /* "head", "live" revision */
```
So I don't see any problem with this change for that.

It doesn't enter the if condition 

```
 if first > NEXT_SNAP:
    log.warning(f"found {o.key}:{dnk} first (0x{first:x}) > NEXT_SNAP (0x{NEXT_SNAP:x})")
    ...
```

Reply:
--repair-nosnap is specifically for first== CEPH_NOSNAP (head) corruption, however, the check is first > NEXT_SNAP. I think this fix still makes sense.

Reply:
> --repair-nosnap is specifically for first== CEPH_NOSNAP (head) corruption, however, the check is first > NEXT_SNAP. I think this fix still makes sense.

Yes, that's right ideally it should have been 
```
if first == CEPH_NOSNAP or first > NEXT_SNAP:
```

It would still work because the first would be a very large unsigned number if it's equal to -2 (i.e., head/CEPH_NOSNAP)